Servers/UI/OJS.Servers.Ui/ClientApp/src/components/common/ChangePageTitle.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+
+import { PageTitles } from '../../common/page-titles';
+
+export const ChangePageTitle = (title: string) => {
+    useEffect(
+        () => {
+            if (title === PageTitles.default) {
+                document.title = PageTitles.default;
+            } else {
+                document.title = `${title} - ${PageTitles.default}`;
+            }
+        },
+        [ title ],
+    );
+
+    return null;
+};
+
+export default { ChangePageTitle };
```

Comment:
This can be achieved with a combination of component wrapper, much like `setLayout` and a hook.

This way, static titles can be defined in `PageContent` as follows:
```
...

const routes = [
    {
        path: '/login',
        Element: withTitle(LoginPage, 'Login in SoftUni judge'),
    },
...
    {
        path: '/submissions',
        Element: withTitle(SubmissionsPage, 'Your submissions - SoftUni Judge'),
    },
...
```

And dynamic titles, i.e. `Problem`, `Submission`, etc, the hook will be used.
```
const SubmissionsPage = (...) => {
    setTitle('...');
};